### PR TITLE
feat: Support Silicate

### LINKED ISSUE #5: feat: Silicate Integration
Eventually, I think we should switch to [Silicate](https://github.com/ModGardenEvent/silicate) for the implementation of the `PredicateBarrierBlock`. This will offer greater flexibility and compatibility as well as make the process a lot easier for developers and datapackers. The only downside is that we'll have yet another library to maintain, but given its architectural nature, Silicate should be relatively easy to maintain as long as there are no extreme Registry refactors.

### PR DESCRIPTION
**Pull request by sylv256**
Fixes #5

### CODE REVIEW

**Review by ChrysanthCow — APPROVED** [review]
Looks good to me! 👍

File: common/src/main/java/net/modgarden/barricade/block/AdvancedBarrierBlock.java
Comment (sylv256) [inline common/src/main/java/net/modgarden/barricade/block/AdvancedBarrierBlock.java:1]: This new system has been tested to work with less bugs.

File: common/src/main/java/net/modgarden/barricade/block/PredicateBarrierBlock.java
Comment (sylv256) [inline common/src/main/java/net/modgarden/barricade/block/PredicateBarrierBlock.java:1]: This new system has been tested to work with less bugs.

**Review by sylv256 — COMMENTED** [review]
why won't it let me approve my own changes 😭 
anyway LGTM (might be one bug i'll fix later idk)

**Pull request merged**